tools/azsdk-cli/docs/specs/7-repo-tooling-contract.md

```diff
@@ -0,0 +1,342 @@
+
+### Current State
+
+- Scripts can be referenced directly in `azsdk` by a hardcoded file path in the process helper [example](https://github.com/Azure/azure-sdk-tools/blob/826bd97dc1580e65b82ac7e4c2c4cd35a1ff8c3a/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs#L54)
+- There are no CI tests in the target repositories to prevent breaking changes being made to referenced scripts.
+- There is no formal versioning/dependency between `azsdk` and the repository state.
+
+### Why This Matters
+
+There is years worth of hardened automation that has been driving the engineering system in the language repositories.
+As the `azsdk` tool is being developed it offers opportunities to streamline some of this automation and also make it accessible
+to AI agents.
+It seems likely that existing repository scripts will always exist in some capacity (see below), so a formalized approach to referencing
+these scripts will have the following benefits:
+
+- Improved velocity - tactical script changes can be made with confidence while not requiring an `azsdk` release.
+  - As `azsdk` matures and gains usage/impact, the validations/requirements to release a new version will increase.
+  - Small, language-specific changes can be made without needing to understand `azsdk` development.
+  - Breakglass changes can be made to resolve urgent issues (unblocking releases, security issues, etc.)
+- Easier migration of scripts to `azsdk`.
+  - It will take time to centralize automation into `azsdk`. There needs to be a bridge between `azsdk` and existing scripts that can support a gradual migration while maintaining reliability.
+- New automation experiments can be made in language repositories, even if the desired end state is a generic interface in `azsdk`.
+- Language specific edge cases can exist without cluttering the cli/mcp tool list.
+- Depending on calling a script (i.e. Export-Apis.ps1) or command line (msbuild /p:ExportApis) that have a set of parameters unique to our repo will be fragile and make versioning `azsdk` with the source changes in the repo very difficult. It will also make this tool very difficult to use in other repo contexts (i.e. the openai or mcp repos).
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Simplify SDK automation
+  - Make automation entrypoints easier to understand and invoke
+  - Invoke operations in the same way locally and in pipelines. Remove multiple entrypoints for the same logic.
+  - Reduce layers of indirection and complexity
+  - Improve testability of changes
+- [ ] Reduce tight coupling between `azsdk` and language repository automation
```

Comment:
I don't think this proposal solves this problem. It's just adding another layer of coupling that's actually harder to understand.

I also don't think this proposal considers any of the nuance with the tools. Some tools cannot run outside of our repos, such as the validate tool. We have no idea what CI checks other repos do, how can we write prompts for that even if we did allow someone to swap in a script that runs their CIs?